sky/clouds/aws.py

```diff
@@ -643,3 +643,47 @@ def _get_disk_specs(cls, disk_tier: Optional[str]) -> Dict[str, Any]:
             'disk_throughput': tier2iops[tier] // 16,
             'custom_disk_perf': tier != 'low',
         }
+
+    @classmethod
+    def check_quota_not_zero(
+        cls,
+        region: str,
+        instance_type: str,
+        use_spot: bool = False) -> bool:
+
+        try:
+            # pylint: disable=import-outside-toplevel,unused-import
+            import boto3
+            import botocore
+            import pandas as pd
+        except ImportError:
+            raise ImportError('Fail to import dependencies for AWS.'
+                              'Try pip install "skypilot[aws]"') from None
+        
+        from sky.clouds.service_catalog import aws_catalog
+        
+        spot_header = ''
+        if (use_spot):
+            spot_header = 'SpotInstanceCode'
+        else:
+            spot_header = 'OnDemandInstanceCode' 
+        
+        quota_code = aws_catalog.get_quota_code(instance_type, spot_header)
+
+        if (quota_code == None):
+            return True
```

Comment:
Would be nice to leave a comment before every return statement to explain why this exit is being taken (e.g. here something like: `# Quota code not found in the catalog for the chosen instance_type, try provisioning anyway`)